### PR TITLE
Added community icon indicator when posting is restricted to mods

### PR DESCRIPTION
**Pull request by hjiangsu**
## Pull Request Description

This PR adds a lock indicator to the community header indicating that posting is restricted to mods. I've also added a tooltip which explains what the lock icon is. If you press and hold on the lock icon, it will show the tooltip. I also updated the localizations to be in alphabetical order.

Note that the create post icon is still enabled as we don't have mechanisms yet to check if the current user has permissions to post. This will most likely be a separate feature in the future!

## Issue Being Fixed

Issue Number: #1022

## Screenshots / Recordings

| | |
|-|-|
| IMG_1565 |  IMG_1566 |

## Checklist

- [x] Did you update CHANGELOG.md?
- [x] Did you use localized strings where applicable?
- [x] Did you add `semanticLabel`s where applicable for accessibility?

### CODE REVIEW

File: lib/l10n/app_en.arb [hunk 1]
```diff
@@ -671,6 +667,10 @@
   "@postViewType": {
     "description": "Label for the post view type (compact/card) in Setting -> Appearance -> Posts"
   },
+  "postingRestrictedToMods": "Posting restricted to mods",
```
Comment (micahmo) [inline lib/l10n/app_en.arb:670]: We do have this string as well!

https://github.com/thunder-app/thunder/blob/c16848ba2768308fb4956c99ac6bfc86ecdbd853/lib/l10n/app_en.arb#L594

**Review by micahmo — APPROVED** [review]
LGTM, pending the ongoing UI discussions!

**Comment by hjiangsu** [comment]
@micahmo This should be good to go now - Added some more screenshots for the light/dark modes!

| | |
|-|-|
|simulator_screenshot_E0081CCF-4681-4047-9365-1E4A54FBD9F0 | simulator_screenshot_47F7CDD7-71B6-40F2-868E-10EA4BD9D4E5|
| simulator_screenshot_5AA698B4-02C3-46FC-93EE-F809042731F9 | simulator_screenshot_E3AD7A1E-DD36-4AB3-BDF4-8EEC467FF679|

**Review by micahmo — APPROVED** [review]
LGTM!

**Pull request merged**